src/codeu/chat/util/TransactionLog.java

```diff
@@ -0,0 +1,100 @@
+package codeu.chat.util;
+
+import codeu.chat.server.Controller;
+import codeu.chat.server.Server;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+
+/**
+ * Utility class for writing and reading to/from the log
+ */
+public class TransactionLog {
+
+  private static final String USER = "ADD-USER";
+  private static final String MESSAGE = "ADD-MESSAGE";
+  private static final String CONVERSATION = "ADD-CONVERSATION";
+  private static final String SPACE = " ";
+
+  public static void writeLog(String type, Object[] params) {
+    String log = "";
+    switch(type) {
+      case "user":
+        log = USER + SPACE +
```

Comment:
you could make this slightly cleaner. have a static map<type, message> to replace the switch clause. use a loop to join the parameters (or use StringUtils.join https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html)